From: Matthew Garrett <mjg@redhat.com>
Date: Mon, 6 Jun 2011 15:36:24 -0400
Subject: [PATCH] x86, efi: Fix argument types for SetVariable()
Git-commit: f7a2d73fe75c71941fb0a6b4d8fe7da8144f2c7b
Patch-mainline: v3.1-rc1
Reference: fate#314499, fate#314500
Target: sle11-sp3

The spec says this takes uint32 for attributes, not uintn.

Signed-off-by: Matthew Garrett <mjg@redhat.com>
Link: http://lkml.kernel.org/r/1307388985-7852-1-git-send-email-mjg@redhat.com
Signed-off-by: H. Peter Anvin <hpa@linux.intel.com>
Acked-by: Lee, Chun-Yi <jlee@suse.com>

---
 arch/x86/platform/efi/efi.c |    2 +-
 include/linux/efi.h         |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

--- a/arch/x86/platform/efi/efi.c
+++ b/arch/x86/platform/efi/efi.c
@@ -124,7 +124,7 @@ static efi_status_t virt_efi_get_next_va
 
 static efi_status_t virt_efi_set_variable(efi_char16_t *name,
 					  efi_guid_t *vendor,
-					  unsigned long attr,
+					  u32 attr,
 					  unsigned long data_size,
 					  void *data)
 {
--- a/include/linux/efi.h
+++ b/include/linux/efi.h
@@ -233,7 +233,7 @@ typedef efi_status_t efi_get_variable_t
 typedef efi_status_t efi_get_next_variable_t (unsigned long *name_size, efi_char16_t *name,
 					      efi_guid_t *vendor);
 typedef efi_status_t efi_set_variable_t (efi_char16_t *name, efi_guid_t *vendor, 
-					 unsigned long attr, unsigned long data_size, 
+					 u32 attr, unsigned long data_size,
 					 void *data);
 typedef efi_status_t efi_get_next_high_mono_count_t (u32 *count);
 typedef void efi_reset_system_t (int reset_type, efi_status_t status,
